Model/Product/Builder.php

```diff
@@ -135,6 +142,42 @@ public function build(Product $product, StoreInterface $store)
     }
 
     /**
+     * Helper method to fetch and return the normalised rating value for a product. The rating is
+     * normalised to a 0-5 value.
+     *
+     * @param Product $product the product whose rating value to fetch
+     * @param StoreInterface $store the store scope in which to fetch the rating
+     * @return float the normalized rating value of the product
+     */
+    private function buildRatingValue(Product $product, StoreInterface $store) {
+        /** @noinspection PhpUndefinedMethodInspection */
+        if (!$product->getRatingSummary()) {
+            $this->reviewFactory->create()->getEntitySummary($product, $store->getId());
```

Comment:
@oostman not really creating an empty entry. This is just following M2 factory pattern where you don't instantiate objects directly but use factory.